homeassistant/core.py

```diff
@@ -2015,7 +2015,7 @@ async def async_store(self) -> None:
             "latitude": self.latitude,
             "longitude": self.longitude,
             "elevation": self.elevation,
-            "unit_system": self.units.name,
+            "unit_system": self.units._name,  # pylint: disable=protected-access
```

Comment:
IMO, `core` is the only place where we should allow access to this property.
I'm not sure if there is a better than using the private attribute.

Reply:
I think it's OK. Maybe add a comment to explain why we have chosen to do this.

Reply:
👍 done